holidays/countries/cocos_islands.py

```diff
@@ -0,0 +1,210 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from datetime import date
+from gettext import gettext as tr
+
+from holidays.calendars import _CustomIslamicHolidays
+from holidays.constants import (
+    JAN,
+    FEB,
+    MAR,
+    APR,
+    MAY,
+    JUN,
+    JUL,
+    AUG,
+    SEP,
+    OCT,
+    NOV,
+    DEC,
+    MON,
+    SAT,
+    SUN,
+)
+from holidays.groups import ChristianHolidays, InternationalHolidays, IslamicHolidays
+from holidays.observed_holiday_base import (
+    ObservedHolidayBase,
+    MON_TO_NEXT_TUE,
+    SAT_SUN_TO_NEXT_MON,
+    SAT_SUN_TO_NEXT_MON_TUE,
+)
+
+
+class CocosIslands(ObservedHolidayBase, ChristianHolidays, InternationalHolidays, IslamicHolidays):
+    """Cocos (Keeling) Islands holidays.
+
+    References:
+        * <https://en.wikipedia.org/wiki/Public_holidays_in_the_Cocos_(Keeling)_Islands>
+        * <https://www.infrastructure.gov.au/territories-regions-cities/territories/indian-ocean-territories/community-bulletins>
+        * <https://www.infrastructure.gov.au/territories-regions-cities/territories/indian-ocean-territories/gazettes-bulletins>
+        * <https://publicholidays.asia/cocos-keeling-islands/>
+    """
+
+    country = "CC"
+    default_language = "coa_CC"
+    # %s (observed).
+    observed_label = tr("%s (disambut)")
+    supported_languages = ("coa_CC", "en_AU", "en_US")
+    start_year = 1955
+
+    def __init__(self, *args, **kwargs):
+        ChristianHolidays.__init__(self)
+        InternationalHolidays.__init__(self)
+        islamic_show_estimated = kwargs.pop("islamic_show_estimated", True)
+        IslamicHolidays.__init__(
+            self, cls=CocosIslandsIslamicHolidays, show_estimated=islamic_show_estimated
+        )
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        super().__init__(*args, **kwargs)
+
+    def _populate_public_holidays(self):
+        # New Year's Day.
+        name = tr("Hari Tahun Baru")
+        self._add_observed(self._add_new_years_day(name))
+        self._add_observed(self._next_year_new_years_day, name=name)
```

Comment:
```suggestion
        self._add_observed(self._add_new_years_day(tr("Hari Tahun Baru")))
```
No need to touch next year's New Year, as there is no rule that shift observed holiday to the _previous_ day, only to the _following_.